api/stream.go

```diff
@@ -200,17 +200,33 @@ func (r streamRoutes) getStream(c *gin.Context) {
 	tumLiveContext := foundContext.(tools.TUMLiveContext)
 	stream := *tumLiveContext.Stream
 	course := *tumLiveContext.Course
-	c.JSON(http.StatusOK,
-		gin.H{"course": course.Name,
-			"courseID":    course.ID,
-			"streamID":    stream.ID,
-			"name":        stream.Name,
-			"description": stream.Description,
-			"start":       stream.Start,
-			"end":         stream.End,
-			"ingest":      fmt.Sprintf("%sstream?secret=%s", tools.Cfg.IngestBase, stream.StreamKey),
-			"live":        stream.LiveNow,
-			"vod":         stream.Recording})
+
+	if stream.Name == "" {
+		stream.Name = stream.GetName()
+	}
+
+	var response gin.H
+	if tumLiveContext.User != nil && tumLiveContext.UserIsAdmin() {
```

Comment:
The "larger" response should be sent to lecturers of the course as well, not only server admins. Also this needs a comment or two, which different purposes the if and else branches serve.